src/osd/PG.h

```diff
@@ -1345,9 +1345,9 @@ class PG : public DoutPrefixProvider {
 
   bool calc_min_last_complete_ondisk() {
     eversion_t min = last_complete_ondisk;
-    assert(!acting_recovery_backfill.empty());
-    for (set<pg_shard_t>::iterator i = acting_recovery_backfill.begin();
-	 i != acting_recovery_backfill.end();
+    assert(!actingset.empty());
+    for (set<pg_shard_t>::iterator i = actingset.begin();
+	 i != actingset.end();
```

Comment:
This looks wrong to me.  We still need to include async_recovery_targets here since they are basically doing recovery based on pg-logs.  Excluding them when calculate min_last_complete_ondisk() can mean over-trimming of pg-logs of those peers and hence cause unexpected errors (e.g.: https://github.com/ceph/ceph/pull/21580).